elastic/beats
# Greenplum Filebeat Module (Enhancement

## What does this PR do?
This PR will provide a filebeat module for reading Greenplum logs in a manner similar to the postgresql module. 

I'm creating a new module and at least one fileset (Greenplum generates many kinds of logs) to manage Greenplum logs. I want the module to ship with one greedy index pattern per fileset, which can handle any logs the fileset throws at it. 
## Why is it important?
Greenplum logs are helpful for troubleshooting issues like segment underperformance or automated jobs which repeatedly error out. My company gets a lot of value from the postgres logs filebeat module and we want to use filebeat on Greenplum too. 

Make an accessible and easy to use Filebeat module that works for any
## Checklist




## Author's Checklist
FAIL: Greenplum test logs are read successfully. I haven't added a test log file yet.


## How to test this PR locally
Pull the PR down, ensure venv is configured make update, make testsuite

There is still a lot of work to do on making tests, I want to test the grok parsing to make it as fast as possible while being generic and flexible.
## Related issues


- 

## Use cases
You have Greenplum and want to store your logs in an Elasticsearch cluster for analysis and visualization.


## Screenshots



## Logs
